### PR TITLE
Ditch haskell setup action

### PR DESCRIPTION
**Pull request by hasufell**

### CODE REVIEW

**Comment by hasufell** [comment]
This should be ready now.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -21,19 +21,26 @@ jobs:
...
-      with:
-        ghc-version: ${{ matrix.ghc }}
+    - if: runner.os == 'Linux' # this is actually a bug in github images: https://github.com/actions/runner-images/pull/6770
+      name: Fix ghcup dir
+      run: echo GHCUP_INSTALL_BASE_PREFIX=/usr/local >> $GITHUB_ENV
+    - name: Setup toolchain
+      run: |
+        which ghcup
+        ghcup install cabal recommended
+        ghcup install ghc --set ${{ matrix.ghc }}
```
Comment (andreasabel) [inline .github/workflows/ci.yml:31]: Would be nice if `ghcup` could also install the prerequisites or print a command that could be executed to do so.

As it stands, you might have to continue adding steps to install prerequisites to various ghc versions --- but you might not be scared of that.

**Review by andreasabel — COMMENTED** [review]
I guess the idea behind `haskell/actions/setup` is to install GHC with its prerequisites.
I also thought of ditching this action, since it is currently badly maintained, but in general such an action makes sense to have and use.  
Of course if `ghcup` had a `sudo` mode also installing prerequisites, then it would fully replace the setup action.
I am neutral on this PR, but please go ahead if you think it is the right step to do.

**Comment by Bodigrim** [comment]
Let's wait for dust to settle (e. g., https://github.com/actions/runner-images/pull/6770) before accumulating more workarounds.

Ideally I'd prefer to keep `haskell/actions/setup`, I just don't get why it cannot get job done reliably.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (hasufell) [reply .github/workflows/ci.yml:31]: > Would be nice if ghcup could also install the prerequisites

It will never do so.

Comment (hasufell) [reply .github/workflows/ci.yml:31]: > As it stands, you might have to continue adding steps to install prerequisites to various ghc versions --- but you might not be scared of that.

Yes, I'm fine with that. The action doesn't add any value and instead makes it hard to reason about what exactly is installed and from what source.

**Comment by hasufell** [comment]
> Ideally I'd prefer to keep haskell/actions/setup, I just don't get why it cannot get job done reliably.

It is unreliable and over-engineered and it's hard to reason about what bindist it installs and from which source.

There's no benefit.

**Comment by hasufell** [comment]
The dust has settled. I want to merge this soon.

**Comment by Bodigrim** [comment]
I would generally prefer haskell/action, but as long as you are willing to maintain the barebone job, I’m fine with it.

**Review by Bodigrim — APPROVED** [review]

**Pull request merged**